### ISSUE TITLE
OSX build fails since commit 'fd_device: allow fd to be passed through a unix socket'

### ISSUE THREAD

**Issue by jaguarondi** [gsliepen/tinc#248]
Commit #f522393 brakes compilation on OSX:

```
⇒  make
/Library/Developer/CommandLineTools/usr/bin/make  all-recursive
Making all in src
  CC       fd_device.o
make[2]: Circular version.c <- version.c dependency dropped.
  GEN      version_git.h
  CC       version.o
fd_device.c:58:45: error: use of undeclared identifier 'MSG_ERRQUEUE'
        if(msg.msg_flags & (MSG_CTRUNC | MSG_OOB | MSG_ERRQUEUE)) {
                                                   ^
fd_device.c:76:4: warning: format specifies type 'unsigned long' but the argument has type 'socklen_t' (aka 'unsigned int') [-Wformat]
                        cmsgptr->cmsg_len, CMSG_LEN(sizeof(device_fd)));
                        ^~~~~~~~~~~~~~~~~
fd_device.c:128:11: warning: implicit declaration of function 'offsetof' is invalid in C99 [-Wimplicit-function-declaration]
                .size = offsetof(struct sockaddr_un, sun_path) + path_length,
                        ^
fd_device.c:128:20: error: expected expression
                .size = offsetof(struct sockaddr_un, sun_path) + path_length,
                                 ^
fd_device.c:128:40: error: use of undeclared identifier 'sun_path'
                .size = offsetof(struct sockaddr_un, sun_path) + path_length,
                                                     ^
2 warnings and 3 errors generated.
  CCLD     tinc
make[2]: *** [fd_device.o] Error 1
make[2]: *** Waiting for unfinished jobs....
clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

**Comment by leptonyu** [comment]
#238 use this patch.

**Comment by gsliepen** [comment]
Fix in commit d0ae7fa0.
